File: src/chapter08/inheritance/Kreis.java
package chapter08.inheritance;

public class Kreis {
	
	private double radius;

	public Kreis() {
		super();
	}
	
	public double getUmfang() {
		return 2 * radius * Math.PI;
	}
	
	public double getFlaeche() {
		return radius * radius * Math.PI;
	}	
	
	@Override
	public String toString() {
		return "Kreis mit radius " + radius 
				+ ": Umfang=" + getUmfang() 
				+ ", Fläche=" + getFlaeche();
	}
	
	public Kreis(double radius) {
		super();
		this.radius = radius;
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		this.radius = radius;
	}

}

File: lab/04-oo/kreis.py
import math

class Kreis:

    def __init__(self, radius):
        self.radius = radius
    
    def umfang(self):
        return 2 * self.radius * math.pi

    def flaeche(self):
        return self.radius * self.radius * math.pi

    def print(self):
        print("Kreis mit Radius", self.radius)
        print("- Umfang = ", self.umfang())
        print("- Fläche =", self.flaeche())


radien = [2,3,7,15]
for radius in radien:
    k = Kreis(radius)
    k.print()


